src/main/java/frc/robot/subsystems/ColorSubsystem.java

```diff
@@ -0,0 +1,97 @@
+package frc.robot.subsystems;
+
+import java.awt.Color;
+import com.revrobotics.ColorSensorV3;
+import edu.wpi.first.wpilibj.I2C.Port;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+/**
+ * A class to facilitate getting one of four colors from the REVRobotics
+ * ColorSensorV3
+ */
+public class ColorSubsystem extends SubsystemBase {
+  /**
+   * @param hsv The array that is used to contain the output of
+   *            java.awt.Color.RGBtoHSB()
+   */
+  public float[] hsv = new float[3];
+  // The ints used to store the raw ADC output of the ColorSensorV3
+  public int red, green, blue;
+  // The maximum allowed absolute distance from the constant to ensure an accurate
+  // reading
+  final float tolerance = 0.025f;
+  // The constants that correspond to the average hue (H) of the Control Panel
+  // colors.
+  final float kYellow = 0.25f;
+  final float kRed = 0.10f;
+  final float kGreen = 0.35f;
+  final float kCyan = 0.50f;
+  // Used to prevent repeated indexing of hsv[]
+  private float hue;
+  // A number used to store the minimum difference between the measured value and
+  // the constants
+  private float mindiff = 1f;
+  // The absolute difference between the measured value and the constants
+  private float diff;
+  // Used to store the return message
+  String color;
```

Comment:
Why are you storing a return value as a public member?  Neither of those things should be true.  Return values should be local to their method.